pkg/storegateway/indexheader/reader_pool.go

```diff
@@ -298,8 +298,9 @@ func (p *ReaderPool) LoadedBlocks() map[ulid.ULID]int64 {
 
 	blocks := make(map[ulid.ULID]int64, len(p.lazyReaders))
 	for r := range p.lazyReaders {
-		if r.reader != nil {
-			blocks[r.blockID] = r.usedAt.Load() / int64(time.Millisecond)
+		usedAt := r.LoadedLastUse()
+		if usedAt != 0 {
+			blocks[r.blockID] = usedAt / int64(time.Millisecond)
```

Comment:
👍 I've bumped into [a race-condition][1], which this one fixes

[1]: https://github.com/grafana/mimir/actions/runs/9563917980/job/26363513562

Reply:
we agreed that it's ok to pull this out of this PR and I can later rebase and resolve conflicts